cadence/server/public/js/aria.js

```diff
@@ -0,0 +1,197 @@
+	window.addEventListener('beforeunload', () => {
+		eventSource.close();
+	});
+
+	eventSource.onerror = function(event) {
+		eventSource.close();
+		retry *= 2;
+		setTimeout(() => { connectToSSE(); }, retry * 1000);
+	}
+
+	eventSource.addEventListener("title", function(event) {
+		$('#song').text(event.data)
+	})
+	eventSource.addEventListener("artist", function(event) {
+		$('#artist').text(event.data)
+	})
+	eventSource.addEventListener("title" || "artist", function(event) {
+		setAlbumArt()
+	})
+	eventSource.addEventListener("listeners", function(event) {
+		if (event.data == -1) {
+			$('#listeners').html("(stream unreachable)")
+		} else {
+			$('#listeners').html(event.data)
+		}
+	})
+	eventSource.addEventListener("listenurl", function(event) {
+		if (event.data == "-/-") {
+			document.getElementById("stream").src = "";
+			$('#status').html("Disconnected from server.")
+		} else {
+			streamSrcURL = location.protocol + "//" + event.data 
+			document.getElementById("stream").src = streamSrcURL
+			$('#status').html("Connected: <a href='"+ streamSrcURL + "'>" + streamSrcURL + "</a>")
+		}
+	})
+}
+
+function postSearch() {
+	// Create a key 'search' to send in JSON
+	var data = {};
+	data.search = $('#searchInput').val();
+	$.ajax({
+		type: 'POST',
+		url: '/api/search',
+		/* contentType sends application/x-www-form-urlencoded data */
+		contentType: 'application/x-www-form-urlencoded',
+		data: JSON.stringify(data),
+		/* dataType expects a json response */
+		dataType: 'json',
+		success: function(data) {
+			let i = 1;
+			// Create the container table
+			var table = "<table class='table is-striped is-hoverable' id='searchResults'>";
+			if (data === null) {
+				console.log("Search completed.  0 results found.");
+				document.getElementById("requestStatus").innerHTML = "Search completed.  0 results found.";
+				// Encode < and >, for error when placed back into no-results message
+				var input = $('#searchInput').val();
+				input = input.replace(/</g, "&lt;").replace(/>/g, "&gt;");
+				// No-results message
+				table += "<div>Nothing found for search '" + input + "' :(</div>";
+			} else {
+				console.log("Search completed. Results found: " + data.length)
+				document.getElementById("requestStatus").innerHTML = "Search completed. Results found: " + data.length;
+				// Build the results table
+				table += "<thead><tr><th>Artist</th><th>Title</th><th>Availability</th></tr></thead><tbody>"
+				data.forEach(function(song) {
+					table += "<tr><td>" + song.Artist + "</td><td>" + song.Title + "</td><td><button class='button requestButton' data-id='" + escape(song.ID) + "'>REQUEST</button></td></tr>";
+				})
+				table += "</tbody>"
+
+			}
+			table += "</table>";
+			// Put table into results html
+			document.getElementById("searchResults").innerHTML = table;
+		},
+		error: function() {
+			console.log("Error. Could not execute search.");
+		}
+	});
+}
+
+
+// Get currently playing album art
+function setAlbumArt() {
```

Comment:
Why is this function different from line 35?

Reply:
So the answer to this (and maybe some of your other questions in `aria.js`) is frontend JavaScript needs refactor and cleanup.